frame/staking-rewards/src/test/mod.rs

```diff
@@ -43,6 +51,45 @@ fn test_create_reward_pool() {
 	});
 }
 
+#[test]
+fn test_stake() {
```

Comment:
We need another test that actually has dilution protection logic kicks in

Reply:
Could you take more details or a link to this check?

Reply:
Make a test that has non zero dilution protection amount.

Reply:
I added the test for this case.